### PR TITLE
Init quantization backend config for inductor

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97122
* #97133
* #97132
* #96475
* #96927
* __->__ #96476

**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
@jerryzh168 @supriyar @vkuzo Could you help to take a look of this PR?

**Review by EikanWang — APPROVED** [review]

File: test/quantization/fx/test_quantize_pt2e.py [hunk 1]
```diff
@@ -190,6 +192,74 @@ def forward(self, x):
             code_after_recompile = m.code
             self.assertTrue(code_before_recompile == code_after_recompile, error_msg)
 
+    @skipIfNoX86
+    @xfailIfPython311
+    def test_inductor_backend_config_conv(self):
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_pt2e.py:197]: can you create a new test case for this test? e.g. TestQuantizePT2EOneDNN

File: test/quantization/fx/test_quantize_pt2e.py [hunk 2]
```diff
@@ -190,6 +192,74 @@ def forward(self, x):
...
+                    export_module, guards = torchdynamo.export(
+                        m,
+                        *copy.deepcopy(example_inputs),
+                        aten_graph=True,
+                        tracing_mode="real",
+                    )
+
+                    qconfig = get_default_qconfig("x86")
+                    qconfig_mapping = QConfigMapping().set_global(qconfig)
+                    backend_config = get_inductor_pt2e_backend_config()
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_pt2e.py:228]: nit: maybe get_x86_inductor_pt2e_backend_config

File: torch/ao/quantization/backend_config/_inductor_pt2e.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    # TODO: remove when functionalization is supported in PT2 mode
+    conv_configs.append(
+        BackendPatternConfig((torch.ops.aten.convolution.default, torch.ops.aten.relu_.default))
+        .set_observation_type(observation_type)  # noqa: E131
+        .set_dtype_configs(dtype_configs)
+        ._set_input_type_to_index({"weight": 1, "bias": 2})
+    )
+    return conv_configs
+
+def get_inductor_pt2e_backend_config():
```
Comment (jerryzh168) [inline torch/ao/quantization/backend_config/_inductor_pt2e.py:41]: nit: maybe get_x86_inductor_pt2e_backend_config

**Review by jerryzh168 — APPROVED** [review]
LGTM, had some nit comments inline

File: test/quantization/fx/test_quantize_pt2e.py (hunk 1 above)
Comment (leslie-fang-intel) [reply test/quantization/fx/test_quantize_pt2e.py:197]: Done.

File: torch/ao/quantization/backend_config/_inductor_pt2e.py
[the remark this answers is not in the document]
Comment (leslie-fang-intel) [reply torch/ao/quantization/backend_config/_inductor_pt2e.py:1]: Thanks for the suggestions. Done.

File: torch/ao/quantization/backend_config/_inductor_pt2e.py (hunk 3 above)
Comment (leslie-fang-intel) [reply torch/ao/quantization/backend_config/_inductor_pt2e.py:41]: Thanks and Done.

**Comment by leslie-fang-intel** [comment]
> LGTM, had some nit comments inline

Thanks for the comments and changed accordingly.

**Pull request closed without merging**